### ISSUE TITLE
BioPerl installation fails because of 1 failed test

### ISSUE THREAD

**Issue by hectorpuigdo** [bioperl/bioperl-live#195]
Hello, everyone:

Yesterday I was trying to install the last version of BioPerl available —1.6.924— when I came across a failed installation. When I looked at the log file (as I tried to install BioPerl with cpanm), I saw that only 1 test had failed:
>Test Summary Report
-------------------
t/Tools/Alignment/Consed.t           (Wstat: 256 Tests: 15 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
Files=334, Tests=18728, 98 wallclock secs ( 2.59 usr  1.09 sys + 77.68 cusr  8.61 csys = 89.97 CPU)
Result: FAIL
Failed 1/334 test programs. 1/18728 subtests failed.

Looking up in the log file, I found this:

>Failed test 'use Bio::Tools::Alignment::Consed;'
at t/Tools/Alignment/Consed.t line 19.
Tried to use 'Bio::Tools::Alignment::Consed'.
Error:  Can't redeclare "my" in "my" at Bio/Tools/Alignment/Consed.pm line 1712, at end of line
Compilation failed in require at t/Tools/Alignment/Consed.t line 19.
BEGIN failed--compilation aborted at t/Tools/Alignment/Consed.t line 19.
Looks like you failed 1 test of 15.
t/Tools/Alignment/Consed.t ............. 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/15 subtests 

Therefore, I went to Bio/Tools/Alignment/Consed.pm line 1712, and this is what it looks like:
`my ($r_source,my $r_destination) = @_;`

So, in fact, there is a double "my" declaration. 

I forced BioPerl installation afterwards and had no problems, but I post this here so in case this is an actual error you can easily fix it for the next version.

Kind regards,
Héctor Puigdomènech Gómez

**Comment by hlapp** [comment]
This was fixed in d13a5ffd9dcaabc3ca09149f657b6aae8fdf8521, but it seems not backported to the 1.6.x release series.

FWIW, a release off of the `master` branch is impending. If you don't strictly need the 1.6.x series, I'd install Bioperl the same way the `latest` version of the Docker container does it, which is right off of Github:

```sh
$ cpanm -v https://github.com/bioperl/bioperl-live/archive/master.tar.gz
```

@cjfields should this be backported to 1.6.x? What puzzles me is that the stable build of the Docker container off of the 1.6.924 CPAN release doesn't throw any error. Perhaps that's because of the version of Perl it gets built with?

**Comment by hectorpuigdo** [comment]
Installing this version worked perfectly. Thank you very much!

**Comment by cjfields** [comment]
@hlapp no need to pull into `v1.6.x`, that branch is now a dead-end release branch and we'll release off `master` from this point on to keep things simple.  

It's possible there were changes on `master` that triggered the test fail or unveiled a bug somehow, and these weren't merged over to `v1.6.x` so tests there pass.  We would need to look into the logs or run `bisect` to work it out, but no need if the bug has been addressed.

BTW, I'm waiting out the NCBI switchover to 'accession-only' data to see whether there is any fallout (should be Sept. 15).  We're in a tricky spot, as they also switch to HTTPS on Sept. 1; one would think they could coordinate this a little more 😄
